scripts/zones/Rabao/npcs/Irmilant.lua

```diff
@@ -3,73 +3,60 @@
-        player:startEvent(135); --Dialogue for those who have completed Indomitable Spirit
+    if player:hasItem(489) == true and (ImmortalLuShang == QUEST_AVAILABLE or ImmortalLuShang == QUEST_COMPLETED) then
+        player:startEvent(77) --Offer the quest if the player has the broken rod
+    elseif player:hasKeyItem(1977) == true and (Indomitable == QUEST_AVAILABLE or Indomitable == QUEST_COMPLETED) then
+        player:startEvent(131) --Begins Indomitable Spirit
+    elseif (Indomitable == QUEST_ACCEPTED or Indomitable == QUEST_COMPLETED) and realday == player:getVar("IndomitableSpiritVar") then
+        player:startEvent(133) --Asks the player to wait
+    elseif (Indomitable == QUEST_ACCEPTED or Indomitable == QUEST_COMPLETED) and realday ~= player:getVar("IndomitableSpiritVar") then
+        player:startEvent(134) --Ends the Quest
+    elseif Indomitable == QUEST_COMPLETED then
+        player:startEvent(135) --Dialogue for those who have completed Indomitable Spirit
     else
-        player:startEvent(76); --Standard Dialogue
+        player:startEvent(76) --Standard Dialogue
     end
-end;
+end
 
 function onEventUpdate(player,csid,option)
-end;
+end
 
 function onEventFinish(player,csid,option)
-    if (csid == 77) then
-        player:addQuest(OUTLANDS,THE_IMMORTAL_LU_SHANG);
-    elseif (csid == 78) then
-        player:tradeComplete();
-        player:addTitle(265);
-        player:addItem(17386); -- Lu Shang's Fishing Rod
-        player:messageSpecial(ITEM_OBTAINED, 17386);
-        player:completeQuest(OUTLANDS,THE_IMMORTAL_LU_SHANG);
-        player:addFame(RABAO,60);
-    elseif (csid == 131) then
-        player:addQuest(OUTLANDS,201);
-    elseif (csid == 132) then
-        player:tradeComplete();
-        player:setVar("IndomitableSpiritVar", os.date("%j")); -- %M for next minute, %j for next day
-    elseif (csid == 134) then
-        if (player:getFreeSlotsCount() == 0) then
-            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED,17011);
-        else
-            player:addTitle(265);
-            player:addItem(17011); -- Ebisu Fishing Rod
-            player:messageSpecial(ITEM_OBTAINED, 17011);
-            player:completeQuest(OUTLANDS,201);
-            player:addFame(RABAO,100);
-        end
+    if csid == 77 then
+        player:addQuest(OUTLANDS,THE_IMMORTAL_LU_SHANG)
+    elseif csid == 78 and npcUtil.completeQuest(player, OUTLANDS, THE_IMMORTAL_LU_SHANG, {item=17386, fameArea=RABAO, fame=60, title=dsp.title.THE_IMMORTAL_FISHER_LU_SHANG}) then
+        player:confirmTrade()
+    elseif csid == 131 then
+        player:addQuest(OUTLANDS, INDOMITABLE_SPIRIT)
+    elseif csid == 132 then
+        player:confirmTrade()
+        player:setVar("IndomitableSpiritVar", os.date("%j")) -- %M for next minute, %j for next day
+    elseif csid == 134 and npcUtil.completeQuest(player, OUTLANDS, INDOMITABLE_SPIRIT, {item=17011, fameArea=RABAO, fame=100, title=dsp.title.INDOMITABLE_FISHER }) then
```

Comment:
Looks like event 134 is called from onTrigger rather than onTrade, so there's no trade to confirm here.  Can just be
```
elseif csid == 134 then
    npcUtil.completeQuest(player, OUTLANDS, INDOMITABLE_SPIRIT, {item=17011, fameArea=RABAO, fame=100, title=dsp.title.INDOMITABLE_FISHER})
end
```